plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -6273,6 +6274,51 @@ public void testDeleteRetainsTableHistory()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    private long getLatestSnapshotId(String tableName)
+    {
+        return (long) computeScalar(format("SELECT snapshot_id FROM \"%s$snapshots\" ORDER BY committed_at DESC FETCH FIRST 1 ROW WITH TIES", tableName));
+    }
+
+    @Test
+    public void testCreateOrReplaceTableSnapshots()
+    {
+        String tableName = "test_create_or_replace_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT BIGINT '42' a, DOUBLE '-38.5' b", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + tableName))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+
+        long v1SnapshotId = getLatestSnapshotId(tableName);
+
+        assertUpdate("CREATE OR REPLACE TABLE " + tableName + " AS SELECT BIGINT '-42' a, DOUBLE '38.5' b", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + tableName))
+                .matches("VALUES (BIGINT '-42', 385e-1)");
+
+        assertThat(query("SELECT a, b  FROM " + tableName + " FOR VERSION AS OF " + v1SnapshotId))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
```

Comment:
Add a test verifying that an initially partitioned table can be replaced with a non-partitioned table.

Reply:
Consider whether it would be relevant to test this scenario in `BaseIcebergSystemTables` as well.

Reply:
It seems that $partitions table is always queryable.

```
trino> create table datalake.default.test as select 1 a;
CREATE TABLE: 1 row

Query 20230907_113254_00002_wg6dm, FINISHED, 1 node
Splits: 18 total, 18 done (100,00%)
0,53 [0 rows, 0B] [0 rows/s, 0B/s]

trino> select * from datalake.default."test$partitions";
 record_count | file_count | total_size |                       data           >
--------------+------------+------------+-------------------------------------->
            1 |          1 |        304 | {a={min=1, max=1, null_count=0, nan_c>
(1 row)
```